src/st/st-widget.c

```diff
@@ -1540,6 +1563,61 @@ on_transition_completed (StThemeNodeTransition *transition,
   st_widget_remove_transition (widget);
 }
 
+static void st_widget_add_background_effects (StWidget    *widget,
+                                              StThemeNode *old_theme_node )
+{
+StThemeNode *new_theme_node = st_widget_get_theme_node (widget);
+
+  if (old_theme_node)
+    {
+      if (old_theme_node->background_blur > 0)
+        {
+          if (widget->priv->background_blur_effect != NULL)
+            {
+              g_object_run_dispose (G_OBJECT (widget->priv->background_blur_effect));
+              g_object_unref (widget->priv->background_blur_effect);
+              widget->priv->background_blur_effect = NULL;
+            }
+        }
+      if (old_theme_node->background_bumpmap != NULL)
+        {
+          if (widget->priv->background_bumpmap_effect != NULL)
+            {
+              g_object_run_dispose (G_OBJECT (widget->priv->background_bumpmap_effect));
+              g_object_unref (widget->priv->background_bumpmap_effect);
+              widget->priv->background_bumpmap_effect = NULL;
+            }
+        }
+    }
+
+  if (new_theme_node)
+    {
+      if (new_theme_node->background_blur > 0)
+        {
+          if (widget->priv->background_blur_effect == NULL)
+            {
+              widget->priv->background_blur_effect = (StBackgroundBlurEffect *) st_background_blur_effect_new ();
+              widget->priv->background_blur_effect->blur_size = new_theme_node->background_blur;
+              for (int i=0;i<4;i++)
+                widget->priv->background_blur_effect->border_radius[i] = new_theme_node->border_radius[i];
+            }
+        }
+      if (new_theme_node->background_bumpmap != NULL)
+        {
+          if (widget->priv->background_bumpmap_effect == NULL)
+            {
+              const char *bumpmap_path;
+
+              widget->priv->background_bumpmap_effect = (StBackgroundBumpmapEffect *) st_background_bumpmap_effect_new ();
+              bumpmap_path = st_theme_node_get_background_bumpmap(new_theme_node);
+              widget->priv->background_bumpmap_effect->bumpmap_path = strdup (bumpmap_path);
+              for (int i=0;i<4;i++)
```

Comment:
Spaces here will make the code easier to read.

Reply:
OK